### ISSUE TITLE
Nested Routes

### ISSUE THREAD

**Issue by brillout** [brillout/vite-plugin-ssr#62]

**Comment by brillout** [comment]
@gryphonmyers FYI. Docs are already live https://github.com/brillout/vite-plugin-ssr#nested-routes.

**Comment by brillout** [comment]
The only thing missing implementation-wise is the `keep-scroll-position` option.

**Comment by gryphonmyers** [comment]
Oh hey sorry when I brought this up before I should have given more context. This is the feature I'm thinking of: https://next.router.vuejs.org/guide/essentials/nested-routes.html Basically it's allowing you to create routes that target deeply nested components. This can be useful for configuration panels and whatnot where you have the same outer component with multiple possible subcomponents, all with their own distinct routes. 

I think I mentioned elsewhere that I'm not terribly impressed with Nuxt's solution for modeling this kind of routing relationship: https://nuxtjs.org/examples/routing-nested-pages/ I found it highly confusing to work with and I would not suggest we do anything like this. 

I'm still not 100% sure what level of support for nested routes is right for this project... all I know is it's a handy feature of Vue Router currently isn't really supported by our vue router integration. It's possible that this doesn't make sense for general-purpose plugin, but perhaps the Vue Router integration could add it as a feature (perhaps by allowing page files to export a parent route by name). 

All in all though, I still haven't dedicated a huge amount of brainpower to this feature.

**Comment by brillout** [comment]
Yes, we do speak about the same thing. I improved the docs to clarify what "Nested Routes" means.

The `vite-plugin-ssr` way to do nested routing is basically:
 - Define a Route String with multiple arguments e.g. `/product/:productId/:productView` (`:productId` determines the outer component while `:productView` determines the inner component).
   ```
   /product/42/pricing                   /product/42/reviews
   +------------------+                  +-----------------+
   | Product          |                  | Product         |
   | +--------------+ |                  | +-------------+ |
   | | Pricing      | |  +------------>  | | Reviews     | |
   | |              | |                  | |             | |
   | +--------------+ |                  | +-------------+ |
   +------------------+                  +-----------------+
   ```
- Use Client-side Routing
- Use `keep-scroll-position`, e.g. `<a href="/product/42/reviews" keep-scroll-position="true" />`.

Would it be possible to pass `:productId` as parent route and `:productView` as child route to Vue Router?

**Comment by brillout** [comment]
Released in `0.1.0-beta.43`.
